minor version update + major edits

kcodes0 opened the pull request:
added:

- beamWidth control to the tool input schema (beam_search)
- numSimulations control to the tool input schema (mcts)

kcodes0 committed:
Update README.md

I added some tests and more a "bloggy" approach to explaining how the reasoning works under the hood I guess?

kcodes0 committed:
Update README.md

added current version, so we can do "what changed" if needed

kcodes0 committed:
added beamWidth control and simulation control to both beam search and MCTS respectively

kcodes0 committed:
Merge pull request #1 from frgmt0/patch-1

Patch 1

kcodes0 committed:
updated readme

kcodes0 committed:
updated version

Jacck commented:
I've found a potential issue while testing: the numSimulations parameter is accepted but the MCTS strategy doesn't seem to reflect the changed value in its metrics (still shows default 50 simulations). Could you check if the simulation count is actually being applied in the MCTS implementation?

Otherwise the implementation looks clean with good input validation.

Regarding benchmark, lets have a talk, I share my email: jack.wojcieszynski@gmail.com

frgmt0 commented:
That is definitely weird. I checked right now and here is the input output:

Default settings:
```json
{
  `thought`: `Testing MCTS with default simulations`,
  `strategyType`: `mcts`,
  `thoughtNumber`: 1,
  `totalThoughts`: 3,
  `nextThoughtNeeded`: true
}
{"thoughtNumber":1,"totalThoughts":3,"nextThoughtNeeded":true,"thought":"Testing MCTS with default simulations","nodeId":"660b90de-bd83-46f0-9203-a810f0f5fd99","score":0.395,"strategyUsed":"mcts","stats":{"totalNodes":138,"averageScore":0.4418236714975843,"maxDepth":5,"branchingFactor":0.5434782608695652,"strategyMetrics":{"beam_search":{"name":"BeamSearchStrategy","nodesExplored":138,"averageScore":0.4418236714975843,"maxDepth":5,"beamWidth":3,"activeBeams":2,"totalBeamNodes":3},"mcts":{"name":"MonteCarloTreeSearchStrategy","nodesExplored":138,"averageScore":0.4418236714975843,"maxDepth":5,"simulationDepth":5,"numSimulations":50,"explorationConstant":1.4142135623730951,"totalSimulations":**51**,"active":true}}}}
```

I believe the default is 50 but im not sure why it lands on 51 unless the initial thought/start of the reasoning chain counts as one, which I think is the case


With 100: 
```json
{
  `thought`: `Testing MCTS with 100 simulations`,
  `strategyType`: `mcts`,
  `thoughtNumber`: 1,
  `totalThoughts`: 3,
  `numSimulations`: 100,
  `nextThoughtNeeded`: true
}
{"thoughtNumber":1,"totalThoughts":3,"nextThoughtNeeded":true,"thought":"Testing MCTS with 100 simulations","nodeId":"eec37abc-d605-4a9c-9aeb-844123e7b027","score":0.38833333333333336,"strategyUsed":"mcts","stats":{"totalNodes":144,"averageScore":0.4404861111111108,"maxDepth":5,"branchingFactor":0.5555555555555556,"strategyMetrics":{"beam_search":{"name":"BeamSearchStrategy","nodesExplored":144,"averageScore":0.4404861111111108,"maxDepth":5,"beamWidth":3,"activeBeams":2,"totalBeamNodes":3},"mcts":{"name":"MonteCarloTreeSearchStrategy","nodesExplored":144,"averageScore":0.4404861111111108,"maxDepth":5,"simulationDepth":5,"numSimulations":100,"explorationConstant":1.4142135623730951,"totalSimulations":**101**,"active":true}}}}
```
I wonder if the issue is that it's not reflected in the input?

As far as chatting you can reach me @      frgmt_@frgmt.xyz

kcodes0 committed:
Major Changes:

added 2 experimental reasoning algorithms

cleaned up codebase to reduce redundant files while retaining the functionality

changed metrics to include new experimental reasoning algorithms when they are used

Updated README.md

Updated LICENSE to 2025

kcodes0 committed:
forgot to update the license

kcodes0 committed:
fixed readme for formatting

frgmt0 commented:
this one is more in depth and a major update id say
